armv7m7-imxrt106x/syspage.c

```diff
@@ -436,35 +439,48 @@ void syspage_addEntries(u32 start, u32 sz)
 int syspage_addProg(void *start, void *end, const char *imap, const char *dmap, const char *name)
 {
 	u8 imapID, dmapID;
-	unsigned int pos = 0, len = 0;
+	unsigned int pos, len;
 	u32 progID = syspage_common.progsCnt;
 
 	if ((syspage_getMapID(imap, &imapID) < 0) || (syspage_getMapID(dmap, &dmapID) < 0)) {
 		plostd_printf(ATTR_ERROR, "\nMAPS for %s doesn not exist!\n", name);
 		return -1;
 	}
 
+	len = plostd_strlen(name);
+
+	if (syspage_common.argCnt + 1 + len + 1 > MAX_ARGS_SIZE) {
+		plostd_printf(ATTR_ERROR, "\nMAX_ARGS_SIZE for %s exceeded!\n", name);
+		return -1;
+	}
+
+	for (pos = 0; pos < len; pos++) {
+		if (name[pos] == ';')
+			break;
+	}
+
+	if (pos > MAX_CMDLINE_SIZE) {
+		plostd_printf(ATTR_ERROR, "\nSyspage program %s, name is too long!\n", name);
+		return -1;
+	}
+
```

Comment:
What if `pos == MAX_CMDLINE_SIZE`.
`low_memcpy()` copy `pos chars`.
and 
```
	while (pos < MAX_CMDLINE_SIZE)
		syspage_common.progs[progID].cmdline[pos++] = 0;
```
 not add `\0` at end.

Reply:
Thank You, corrected.